### PR TITLE
Fix fpgabist for PAC regressions

### PR DESCRIPTION
**Pull request by michael-adler**
The Vista Creek (VC) merge left fpgabist broken when used with Darby Creek
regressions. The new VC bist code hard-wired the device ID of Rush Creek and
promoted device ID handling so that it had to match the board, even when a
full BDF was supplied.

The changes here:

- Demote --device-id so it is just one of the components in the search for a board.
  The default value is None. The new algorithm looks for all available FPGAs on the
  host and then filters them using the supplied BDF and device ID. The test can run
  as long as the filter matches a single FPGA.

- Cherry pick a few changes from the DC branch.

- Repair the non-trivial merge of the DC and the VC changes.

- Fix nlb0 clock frequency discovery. It was reporting in correct performance.

### CODE REVIEW

**Comment by michael-adler** [comment]
Test state: http://sw-pert.altera.com/pert/pert.php?test_run_id=3698500

DMA doesn't compile yet due to Qsys, which accounts for a number of the failures.

File: tools/extra/pac/fpgabist/bist_common.py [hunk 1]
```diff
@@ -54,21 +54,26 @@ def check_required_cmds():
...
         output = subprocess.Popen(cmd, stdout=subprocess.PIPE)
         json_data = json.loads(output.communicate()[0])
         accel_data = json_data["afu-image"]["accelerator-clusters"]
         uuid = accel_data[0]["accelerator-type-uuid"].encode("ascii")
         return uuid.lower().replace("-", "")
-    else:
-        bdf_pattern = re.compile(BDF_PATTERN)
-        for fpga in glob.glob('/sys/class/fpga/*'):
+    elif bdf:
+        pattern = '{:02x}:{:02x}.{:x}'.format(bdf['bus'], bdf['device'],
```
Comment (buie1) [inline tools/extra/pac/fpgabist/bist_common.py]: Would it be easier to rename the macro BDF_PATTERN with this pattern you are looking for here?
Did you find an issue using the other pattern or is this just a cleaner regex?

File: tools/extra/pac/fpgabist/bist_common.py [hunk 2]
```diff
@@ -120,11 +125,13 @@ def get_mode_from_path(gbs_path):
     return None
 
 
-def load_gbs(gbs_file, bus_num):
+def load_gbs(gbs_file, bdf):
     print "Attempting Partial Reconfiguration:"
-    cmd = "{} -B 0x{} -v {}".format('fpgaconf', bus_num, gbs_file)
+    cmd = ['fpgaconf', '-B', hex(bdf['bus']), '-D',
```
Comment (buie1) [inline tools/extra/pac/fpgabist/bist_common.py]: This is a good addition here.

File: tools/extra/pac/fpgabist/bist_dma.py [hunk 3]
```diff
@@ -55,26 +55,33 @@ def run_cmd(self, cmd):
...
-            bc.load_gbs(gbs_path, bus_num)
+            bc.load_gbs(gbs_path, bdf)
         ret = 0
         for func, param in self.executables.items():
             if bd_id in dma_list:
                 for i, c in dma_list[bd_id].items():
                     name, size = c
                     if self.dev_id != int(param, 16):
                         continue
-                    cmd = "{} {} -B 0x{} -D {} -S {}".format(func, param,
```
Comment (buie1) [inline tools/extra/pac/fpgabist/bist_dma.py]: Is this working? Removing the leading `0x` before this was required so are you passing the leading `0x` with each element of the bdf?

Comment (michael-adler) [reply tools/extra/pac/fpgabist/bist_dma.py]: No, DMA is broken still. I need the changes that were cherry picked into the internal master to test it, so was waiting for this pull to go in.

File: tools/extra/pac/fpgabist/bist_common.py (hunk 1 above)
Comment (michael-adler) [reply tools/extra/pac/fpgabist/bist_common.py]: Ok. I'll do that in the next round with the DMA update.

**Pull request merged**